pkg/model/deployment_test.go

```diff
@@ -169,6 +170,35 @@ spec:
 	assert.Equal(t, "my-vol", model.backstageDeployment.deployment.Spec.Template.Spec.Volumes[3].Name)
 }
 
+func TestImageInCRPrevailsOnEnvVar(t *testing.T) {
+	bs := *deploymentTestBackstage.DeepCopy()
+	bs.Spec.Deployment = &bsv1.BackstageDeployment{}
+	bs.Spec.Deployment.Patch = &apiextensionsv1.JSON{
+		Raw: []byte(`
+spec:
+ template:
+   spec:
+     containers:
+       - name: backstage-backend
+         image: cr-image
+`),
+	}
+
+	os.Setenv(BackstageImageEnvVar, "envvar-image")
```

Comment:
```suggestion
	t.Setenv(BackstageImageEnvVar, "envvar-image")
```

Recommended to use `t.Setenv` instead to make sure this temporary env var value is unset automatically after the test, so it does not interfere with others trying to use the same environment variable.